pkg/objstore/swift/swift.go

```diff
@@ -87,206 +205,138 @@ func (c *Container) Name() string {
-				return false, err
+		for _, object := range objects {
+			if err := f(object); err != nil {
+				return objects, errors.Wrap(err, "swift iteration over objects")
 			}
 		}
-
-		return true, nil
+		return objects, nil
 	})
 }
 
-// Get returns a reader for the given object name.
-func (c *Container) Get(ctx context.Context, name string) (io.ReadCloser, error) {
+func (c *Container) get(name string, headers swift.Headers, checkHash bool) (io.ReadCloser, error) {
 	if name == "" {
-		return nil, errors.New("error, empty container name passed")
+		return nil, errors.New("Object name cannot be empty")
 	}
-	response := objects.Download(c.client, c.name, name, nil)
-	return response.Body, response.Err
+	file, _, err := c.connection.ObjectOpen(c.name, name, checkHash, headers)
+	if err != nil {
+		return nil, errors.Wrap(err, "swift open object")
+	}
+	return file, err
+}
+
+// Get returns a reader for the given object name.
+func (c *Container) Get(ctx context.Context, name string) (io.ReadCloser, error) {
+	return c.get(name, swift.Headers{}, true)
 }
 
-// GetRange returns a new range reader for the given object name and range.
 func (c *Container) GetRange(ctx context.Context, name string, off, length int64) (io.ReadCloser, error) {
-	options := objects.DownloadOpts{
-		Newest: true,
-		Range:  fmt.Sprintf("bytes=%d-%d", off, off+length-1),
+	// Set Range HTTP header, see the docs https://docs.openstack.org/api-ref/object-store/?expanded=show-container-details-and-list-objects-detail,get-object-content-and-metadata-detail#id76.
+	bytesRange := fmt.Sprintf("bytes=%d-", off)
+	if length != -1 {
+		bytesRange = fmt.Sprintf("%s%d", bytesRange, off+length-1)
 	}
-	response := objects.Download(c.client, c.name, name, options)
-	return response.Body, response.Err
+	return c.get(name, swift.Headers{"Range": bytesRange}, false)
 }
 
 // Attributes returns information about the specified object.
 func (c *Container) Attributes(ctx context.Context, name string) (objstore.ObjectAttributes, error) {
-	response := objects.Get(c.client, c.name, name, nil)
-	headers, err := response.Extract()
+	if name == "" {
+		return objstore.ObjectAttributes{}, errors.New("Object name cannot be empty")
+	}
+	info, _, err := c.connection.Object(c.name, name)
 	if err != nil {
-		return objstore.ObjectAttributes{}, err
+		return objstore.ObjectAttributes{}, errors.Wrap(err, "swift get object attributes")
 	}
-
 	return objstore.ObjectAttributes{
-		Size:         headers.ContentLength,
-		LastModified: headers.LastModified,
+		Size:         info.Bytes,
+		LastModified: info.LastModified,
 	}, nil
 }
 
 // Exists checks if the given object exists.
 func (c *Container) Exists(ctx context.Context, name string) (bool, error) {
-	err := objects.Get(c.client, c.name, name, nil).Err
-	if err == nil {
-		return true, nil
-	}
-
-	if _, ok := err.(gophercloud.ErrDefault404); ok {
-		return false, nil
+	_, _, err := c.connection.Object(c.name, name)
+	if err != nil {
```

Comment:
I would be explicit here (: 

```go
if c.IsObjNotFoundErr(err) {
  return false, nil
}

if err != nil {
  return false, errors.Wrap(err, "swift check if file exists")
}

return true, nil
```

Reply:
bit safer (:

Reply:
less indentation is always better :+1: :)